pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -213,14 +214,29 @@ def preprocess(
         if isinstance(circuits, QuantumScript):
             circuits = [circuits]
             is_single_circuit = True
-
-        if self.wires and (
-            extra_wires := set(Wires.all_wires(c.wires for c in circuits)) - set(self.wires)
-        ):
-            raise WireError(
-                f"Cannot run circuit(s) on {self.name} as they contain wires "
-                f"not found on the device: {extra_wires}"
-            )
+        else:
+            # if it's a tuple, it will break assignment below
+            circuits = list(circuits)
+
+        if self.wires:
+            if extra_wires := set(Wires.all_wires(c.wires for c in circuits)) - set(self.wires):
+                raise WireError(
+                    f"Cannot run circuit(s) on {self.name} as they contain wires "
+                    f"not found on the device: {extra_wires}"
+                )
+            for i, circuit in enumerate(circuits):
+                measurements = circuit.measurements.copy()
+                modified = False
+                for m_idx, mp in enumerate(measurements):
+                    if not mp.obs and not mp.wires:
+                        modified = True
+                        new_mp = copy(mp)
+                        new_mp._wires = self.wires  # pylint:disable=protected-access
```

Comment:
```suggestion
                        new_mp = mp.__class__(wires=self.wires)
```
To avoid in-place mutation.

Reply:
I had this before, but then `CountsMP` didn't preserve the `all_outcomes` kwarg and that was even worse (check commit history for my woes). I'm only in-place-mutating a copy of the MP, so it shouldn't cause any issues. Inspiration to copy-and-mutate came from [here](https://github.com/PennyLaneAI/pennylane/blob/92b8e8390c29d35dbe48927264d65110b8fe8fc5/pennylane/measurements/measurements.py#L445-L449C32), which also needed investigating in this PR 🙃

Reply:
added task [here](https://app.shortcut.com/xanaduai/story/40588/measurementprocess-are-registered-as-pytrees)